code/game/gamemodes/revolution/revsquad.dm

```diff
@@ -0,0 +1,309 @@
+		completion_text = "<br><span class='danger'><FONT size = 3> The heads of staff managed to stop the revolution!</FONT></span>"
+	..()
+	return 1
+
+/datum/game_mode/proc/auto_declare_completion_revsquad()
+	var/list/targets = list()
+	var/text = ""
+	if(head_revolutionaries.len || istype(ticker.mode,/datum/game_mode/revolution))
+		var/icon/logo1 = icon('icons/mob/mob.dmi', "rev_head-logo")
+		end_icons += logo1
+		var/tempstate = end_icons.len
+		text += {"<img src="logo_[tempstate].png"> <FONT size = 2><B>The revolutionary squad members were:</B></FONT> <img src="logo_[tempstate].png">"}
+
+		for(var/datum/mind/headrev in head_revolutionaries)
+			if(headrev.current)
+				var/icon/flat = getFlatIcon(headrev.current, SOUTH, 1, 1)
+				end_icons += flat
+				tempstate = end_icons.len
+				text += {"<br><img src="logo_[tempstate].png"> <b>[headrev.key]</b> was <b>[headrev.name]</b> ("}
+				if(headrev.current.isDead())
+					text += "died"
+					flat.Turn(90)
+					end_icons[tempstate] = flat
+				else if(headrev.current.z != 1)
+					text += "fled the station"
+				else
+					text += "survived the revolution"
+				if(headrev.current.real_name != headrev.name)
+					text += " as [headrev.current.real_name]"
+			else
+				var/icon/sprotch = icon('icons/effects/blood.dmi', "floor1-old")
+				end_icons += sprotch
+				tempstate = end_icons.len
+				text += {"<br><img src="logo_[tempstate].png"> <b>[headrev.key]</b> was <b>[headrev.name]</b> ("}
+				text += "body destroyed"
+			text += ")"
+
+			for(var/datum/objective/mutiny/objective in headrev.objectives)
+				targets |= objective.target
+
+
+	if(revolutionaries.len || istype(ticker.mode,/datum/game_mode/revolution))
+		var/icon/logo2 = icon('icons/mob/mob.dmi', "rev-logo")
+		end_icons += logo2
+		var/tempstate = end_icons.len
+		text += {"<br><img src="logo_[tempstate].png"> <FONT size = 2><B>The recruited revolutionaries were:</B></FONT> <img src="logo_[tempstate].png">"}
+
+		for(var/datum/mind/rev in revolutionaries)
+			if(rev.current)
+				var/icon/flat = getFlatIcon(rev.current, SOUTH, 1, 1)
+				end_icons += flat
+				tempstate = end_icons.len
+				text += {"<br><img src="logo_[tempstate].png"> <b>[rev.key]</b> was <b>[rev.name]</b> ("}
+				if(rev.current.isDead())
+					text += "died"
+					flat.Turn(90)
+					end_icons[tempstate] = flat
+				else if(rev.current.z != 1)
+					text += "fled the station"
+				else
+					text += "survived the revolution"
+				if(rev.current.real_name != rev.name)
+					text += " as [rev.current.real_name]"
+			else
+				var/icon/sprotch = icon('icons/effects/blood.dmi', "floor1-old")
+				end_icons += sprotch
+				tempstate = end_icons.len
+				text += {"<br><img src="logo_[tempstate].png"> <b>[rev.key]</b> was <b>[rev.name]</b> ("}
+				text += "body destroyed"
+			text += ")"
+
+
+
+	if( head_revolutionaries.len || revolutionaries.len || istype(ticker.mode,/datum/game_mode/revolution) )
```

Comment:
I have no idea why this is here
Rev's scoreboard code that I stole this from is pretty awful
I'm taking out the istype